source/dub/compilers/compiler.d

```diff
@@ -108,10 +109,24 @@ interface Compiler {
 	void setTarget(ref BuildSettings settings, in BuildPlatform platform, string targetPath = null) const;
 
 	/// Invokes the compiler using the given flags
-	void invoke(in BuildSettings settings, in BuildPlatform platform, void delegate(int, string) output_callback);
+	deprecated("specify the working directory")
+	final void invoke(in BuildSettings settings, in BuildPlatform platform, void delegate(int, string) output_callback)
+	{
+		invoke(settings, platform, output_callback, getWorkingDirectory());
+	}
+
+	/// ditto
+	void invoke(in BuildSettings settings, in BuildPlatform platform, void delegate(int, string) output_callback, NativePath cwd);
 
 	/// Invokes the underlying linker directly
-	void invokeLinker(in BuildSettings settings, in BuildPlatform platform, string[] objects, void delegate(int, string) output_callback);
+	deprecated("specify the working directory")
+	final void invokeLinker(in BuildSettings settings, in BuildPlatform platform, string[] objects, void delegate(int, string) output_callback)
```

Comment:
Technically this is a breaking change but let's roll with it.

Reply:
yeah I'm not too keen on this change, but it's unlikely third party apps have inherited the compiler class to add their own compilers. If they did, it's an easy fix at least.